docs/ARCHITECTURE.md

```diff
@@ -24,79 +24,386 @@ Before delving into the details, let's understand a few key concepts that will b
+    "erc20bridge.sw" {
+        function process_message
+    }
+    "(user) ERC20 depositor" }|--|| "bridge predicate" : "run"
+    "bridge predicate" ||--|| "erc20bridge.sw" : "relay MessageSent data"
+```
+
+#### L1 (Ethereum) ETH Deposit
+
+Since `MessageSent` events and their spawned `MessageCoin` UTXOs can carry a value, an ETH deposit consists merely of the action of calling `sendMessage` on the L1 with some attached ETH. The recipient in the L2 will see its balance updated by means of the inclusion of the `MessageCoin` that reflects this value in the L2.
+
+```mermaid
+sequenceDiagram
+    participant user as User
+    box purple ETH
+    participant eth as L1 Chain
+    participant portal as FuelMessagePortal.sol
+    end
+
+    box darkgreen FUEL
+    participant sequencers as Fuel Sequencers
+    participant fuel as L2 Chain
+    end
+
+
+    user ->>+ eth: send tx with ETH
+    eth ->> portal: sendMessage()
+    note over eth,portal: {to: fuel address}
+    portal --> eth: emit MessageSent event
+    eth ->>- user: Confirm tx
+    loop MessageSent subscription
+    sequencers ->> portal: get MessageSent events
+    portal ->> sequencers: MessageSent events
+    sequencers ->> fuel: Include events
+    end
+    note over sequencers,fuel: The deposit is reflected as an UTXO<br/>owned by the recipient of the<br> MessageSent event in ETH
+```
+
+#### L1 (Ethereum) ERC20 Deposit
+
+A deposit operation is an use case of the message passing system from L1 entities to L2 entities. The user will send a transaction to deposit an asset in a L1 bridge contract `FuelERC20Gateway`, which will in turn call `sendMessage` and emit a `MessageSent` event through the `FuelMessagePortal` to be picked up by the Fuel blockchain 's sequencer. The recipient of this message shall be the `bridge predicate` (a L2 entity that posseses an ID), whose role will be explained further ahead. The payload of the message will just reflect the fact that the `FuelERC20Gateway` has received a deposit of a L1 token, with a given amount and a desired recipient. Sequencers will include this message as part of an UTXO that can be spent by the `bridge predicate`. The `predicate` holds these UTXOs until any other entity (which can be the users themselves, or another "relayer" entity) executes the predicate 's logic, which has the responsibility for running validation logic before sending the data to the bridge contract on the L2, `erc20bridge.sw`, by calling `process_message`. Once the UTXO is "spent" (delivered), this last contract will have the capability of processing the payload inside the message, validating that their sender is the L1 bridge contract `FuelERC20Gateway`, and then proceeding to mint an equivalent asset and amount in the Fuel blockchain to the one originally deposited in the L1. Then, the minted assets become available to the user in the L2.
+
+```mermaid
+sequenceDiagram
+    participant user as User
+    box purple ETH
```

Comment:
changing this to light yellow makes it much easier to read

Reply:
Changed boxes to be transparent with a frame